### DISCUSSION TITLE
How to specify service name when using opentelemetry-spring-starter

### DISCUSSION THREAD

**Discussion by tinnapat** [open-telemetry/opentelemetry-java-instrumentation#4031] [Q&A] [answered]
Hi

How can I specify service name when using `opentelemetry-spring-starter`?

In the [documents](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/spring/spring-boot-autoconfigure/README.md#features) there is no properties to specify service name. I have tried setting `otel.service.name` in spring configuration and also by setting `OTEL_SERVICE_NAME` environment variables, but the service name shown always be `unknown_service:java`

Thank you very much.

**Comment by mateuszrzeszutek** [comment, marked as answer]
Hey @tinnapat,
Currently it's not really possible to set the service name (or any other resource attribute) in the `opentelemetry-spring-boot-autoconfigure` project. There's already an issue (https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/3480) about it.
